Mathlib/Algebra/Order/Field/Canonical/Defs.lean

```diff
@@ -34,3 +34,32 @@ instance (priority := 100) CanonicallyLinearOrderedSemifield.toCanonicallyLinear
     [CanonicallyLinearOrderedSemifield α] : CanonicallyLinearOrderedAddCommMonoid α :=
   { ‹CanonicallyLinearOrderedSemifield α› with }
 #align canonically_linear_ordered_semifield.to_canonically_linear_ordered_add_monoid CanonicallyLinearOrderedSemifield.toCanonicallyLinearOrderedAddCommMonoid
+
+open Function
+
+-- this makes `mul_lt_mul_left`, `mul_pos` etc work on `ℤₘ₀`
+instance {α : Type*} [Mul α] [Preorder α] [CovariantClass α α (· * ·) (· < ·)]:
+    PosMulStrictMono (WithZero α) where
+  elim := @fun
+    | ⟨(x : α), hx⟩, 0, (b : α), _ => by
+        rw [mul_zero]
+        exact WithZero.zero_lt_coe _
```

Comment:
```suggestion
        simpa only [mul_zero] using WithZero.zero_lt_coe _
```

Reply:
(this is a comment on one of the dependent PRs and was dealt with there)